src/intro-to-c/arrays/arrays-intro.md

```diff
@@ -0,0 +1,300 @@
+Our array currently looks like this now:
+
+```
++------+------+------+------+------+
+| 92.5 |      |      |      |      |
++------+------+------+------+------+
+```
+
+We've populated the first cell with a grade, by using our array access operator. We can do this for the rest of the grades as well:
+
+```c
+grades[0] = 92.5;
+grades[1] = 52.4;
+grades[2] = 75.0;
+grades[3] = 82.0;
+grades[4] = 88.9; // this is the last element we can access in our grades array!
+```
+
+Here is what our array looks like now:
+
+```
++------+------+------+------+------+
+| 92.5 | 52.4 | 75.0 | 82.0 | 88.9 |
++------+------+------+------+------+
+```
+
+You can also overwrite the values in an array by using the array access operator:
+```c
+grades[3] = 23.0 // they failed...
+```
+
+Here's a preview of the array after overwriting what's in index 3 of `grades`:
+```
++------+------+------+------+------+
+| 92.5 | 52.4 | 75.0 | 23.0 | 88.9 |
++------+------+------+------+------+
+```
+
+Now we finally have our array full of grades! Obviously, this is a pretty tedious way to populate our array; we just manually typed out the grades for each student! However, there are other ways in which we can initialize and populate an array:
+
+## Other Ways to Create and Populate an Array
+
+### Initializer List
+If we already know what will be in our array, we can create an array like this, using what is called an initializer list:
+```c
+int favorite_numbers[] = {3, 5, 9, 42, 1, 3141592};
+```
+Notice how we didn't need to specify the size of the array. This is because we used curly brackets (`{}`) to 'initialize' our array inline, so C will automagically create our array with the smallest size needed to house all of our elements. We can verify this by using `sizeof`:
+
+<details>
+<summary>Show code</summary>
+
+```c
+#include <stdio.h>
+int main(void) {
+    // array of integers, with 6 numbers inside of it:
+	int favorite_numbers[] = {3, 5, 9, 42, 1, 3141592};
+    // use sizeof to calculate the size of the array:
+    // you don't need to know how this works for now,
+    // but just know that this formula will give us the size of our array
+	int array_size = sizeof(favorite_numbers) / sizeof(int);
+	printf("length of the array: %d\n", array_size);
+	return 0;
+}
+```
+```
+$ ./a.out
+length of the array: 6
+```
+</details>
+
+Neat tech! Of course, we can still put the size if we wanted, but **be careful** not to overstep the bounds of the array:
+```c
+// array created with initializer list
+float grades[5] = {92.5, 52.4, 75.0, 82.0, 88.9};
+
+// *** DON'T DO THIS! We're overstepping our bounds! ***
+int numbers[3] = {1, 2, 3, 4, 5}; // array of size 3 with 5 elements?! uh oh...
+```
+
+### Loops
+We can also use loops to initialize our array, such as a for loop. Can you guess what our array will look like after running this code?
+
+```c
+int numbers[5];
+for (int x = 0; x < 5; x++) {
+    numbers[x] = x*2;
+}
+```
```

Comment:
Great example here! Just a small tweak, I would suggest changing the `x` to `i` to be in-line with the usual nomenclature used in loops.

Reply:
You're right, I don't even know why I used x lol. I think I wanted it to look like a formula for the part below but I think it makes sense anyways with i